### ISSUE TITLE
Use of inteface type parameter `x` as `...x` causes "Did not expect BoundT"

### ISSUE THREAD

**Issue by Gozala** [facebook/flow#4752]
[Here is an example](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVBLAdgFwKYBOUAhgMZ5gBKcArvgDy04B8YA3qmGFHAQLbEcACgB0YpgEoAXAGccBbAHNUAXyA) illustrating the issue:

```js
/* @flow */

interface Route<out> {
  format(...out):string
}
```

**Comment by asolove** [comment]
Just to be more explicit about why you would want this and what it is expected to mean, I’d love to have some more context. By spreading a type into the arg types of a function, I’m guessing out is a tuple  type and you want the respective elements to be the types of the respective parameters?

**Comment by Gozala** [comment]
> Just to be more explicit about why you would want this and what it is expected to mean, I’d love to have some more context.

I'm working on [type safe routing library](https://github.com/gozala/route.flow), where each route can parse / format URLs. Idea is that parse returns a tuple for the route parameters `Route<out>.parse(URL):out` and the same route can be given typed route parameters to format a URL `Route<out>.format(...out):URL`.

Here's a specific example:

```js
const calculator = Router.Root
  .segment("calculator")
  .param(Route.Float)
  .segment("+")
  .param(Route.Float)
```

Type of the `calculator` is `Route<[float, float]>` so that following will be true

```js
calculator.parse("/calculator/313/+/") //> null
calculator.parse("/calculator/313/+/3") //> [313, 3]
calculator.format(313, 3) //> "/calculator/313/+/3"
```

**Comment by Bannerets** [comment]
Fixed in v0.63.0.
